dnsimple/registrar_delegation.go

```diff
@@ -7,12 +7,28 @@ import (
 // Domain represents a list of name servers that correspond to a domain delegation.
 type Delegation []string
 
+// Vanity Name Server
+type VanityNameServer struct {
```

Comment:
How is this different than a standard `NameServer`? Given we'll have to implement also support for name servers, I suggest to name it `VanityNameServer`.

Also please use a comment that starts with `NameServer` (the exact struct name) or go-lint will complain.